src/types/seeds.ts

```diff
@@ -0,0 +1,119 @@
+
+import { Table } from '../db/db';
+import db from '../db/sql-db'
+
+import { ArtistProfileKeys } from './artist';
+import { getCrdtUpdateMessage, getCrdtUpsertMessage } from './message'
+import { NFTContractTypeName, NFTFactoryKeys, NFTStandard } from './nft';
+import { MusicPlatform, MusicPlatformKeys, MusicPlatformType } from './platform';
+import { ProcessedTrackKeys } from './track';
+
+enum SeedEntities {
+  'platforms',
+  'nftFactories',
+  'artistProfiles',
+  'processedTracks'
+}
+type SeedEntity = keyof typeof SeedEntities;
+
+enum PlatformsRequiredKeys {
+  ID = 'id',
+  NAME = 'name',
+  TYPE = 'type',
+}
+
+enum NFTFactoriesRequiredKeys {
+  ID = 'id',
+  PLATFORM_ID = 'platformId',
+  CONTRACT_TYPE = 'contractType', // validated against `NFTContractTypeName`
+  STANDARD = 'standard', // validated against `NFTStandard`
+  AUTO_APPROVE = 'autoApprove',
+  APPROVED = 'approved'
+}
+
+enum ArtistProfilesRequiredKeys {
```

Comment:
Agree we do need to work out a clear plan re the async nature of seeds/indexing and separation of concerns between them with validation.

**On foreign ids:**

In this case with missing foreign ids, i guess some options that come to mind are:
 - Query the DB to validate their existence as part of validation, and reject
 - Allow them through, let them fail when processed and have no effect
 - Allow them through, but update the trigger that processes messages so that it does a more general filter that filters out messages with missing foreign keys, assuming they're not ready for processing yet but that they may be available later

For this PR, I don't think we need to support artist profiles, just artists, where there shouldn't be any foreign keys _(and so then we'll probably move stuff like avatarURL and websiteURL up to artist instead of artistProfile maybe after this PR)._

**Re the update/upsert:**
Does it work to just have a rule like:
 - if a message is an upsert, all keys are required and validate them all
 - if a message is an update, at least one key is required, so just validate that

Reply:
**Foreign IDS:**
I'm thinking for the sake of closing this PR off before it grows any larger, we go for :
> Allow them through, let them fail when processed and have no effect

We can then follow up with adjusting the filter to be more selective and robust afterwards - especially seeing as using this internally for now and will likely be pretty careful with the json payloads.

**Support for artists vs artistProfiles**
We might have a similar problem with `tracks`, so we'll need to sort this out eventually. I reckon it won't be too complex to add `artists` to this PR as well. I'll give it a go and if it becomes complex, I'll bail and just reduce the scope of artistProfiles to artists.

**Re the update/upsert:**
That should work. I'll attempt a small extraction of one or two of the existing `/migrations` to confirm whether these validation options will work as we need.  They're the closest thing we have to a practical use case for now and I'm sure we'll need to extend validation as we discover more use cases to cover.

Reply:
Cool also keep in mind that artistProfiles is kinda irrelevant once we have spinamp-centric ones - we may even be able to deprecate that entity entirely in future

Reply:
> We might have a similar problem with `tracks`, so we'll need to sort this out eventually. I reckon it won't be too complex to add `artists` to this PR as well. I'll give it a go and if it becomes complex, I'll bail and just reduce the scope of artistProfiles to artists.

yeh for tracks i think we should only allow updates, and should not allow any foreign key fields for now

Reply:
I think I just realised the scope of the problem changes with ArtistProfiles in that we don't have an `id` on it, so updates are a problem inherently. I've yanked it from this PR. 

Agree that we should move or deprecate fields from `ArtistProfile` as a follow up.